Dandelion++-review.md

```diff
@@ -0,0 +1,120 @@
+
+I talked to Reuben Yeap, Co-Founder & Project Steward of the Firo project, the first crypto project that implemented Dandelion++. In his words, he said that the implementation of the Dandelion++ was not that complicated, but annoying at some times. When talking about the technical part of the protocol on Firo he stated:
+
+"The system essentially functions with two distinct mempools: one dedicated to Dandelion++ transactions and the other for the remainder of the transactions. All inventory commands capable of propagating transactions to other hosts possess corresponding versions specifically designed for use during the Dandelion++ stem phase. Subsequently, transactions disseminate in a standard manner through conventional commands."
+
+When talking about the process of implementation he affirmed:
+
+"Our code was adapted from a pull request submitted to the Bitcoin project, which, at the time, was still under consideration as a proposal. We have made several modifications and improvements to it to suit our specific requirements."
+
+(Firo video explaining Dandelion++ https://www.youtube.com/watch?v=jQwEo8O-nE4)
+
+Debates in cryptocurrency forums about Dandelion++ implementation highlight several challenges: integrating it into existing systems can be complex, it may affect network efficiency, achieving community consensus is difficult, and there are concerns about balancing privacy improvements with potential increases in latency.  When coming to Bitcoin the discussion revolved around 3 main topics:
+
+Complexity in DoS Prevention: The Dandelion++ protocol introduces potential DoS vectors not present in the traditional transaction relay model. The acceptance of transactions into the mempool before the relay introduces complexity in managing and preventing bandwidth and CPU-exhaustion DoS attacks. Implementing safeguards against these attacks without significantly complicating the codebase or reducing network efficiency is a major challenge.
+Implementation and Maintenance Overhead: The introduction of a "stempool" or equivalent logic to manage the unique aspects of Dandelion++ transactions adds to the codebase's complexity. This complexity not only pertains to the initial implementation but also to ongoing maintenance, potentially diverting resources from other critical developments.
+Questionable Cost-Benefit Ratio: Considering the implementation's complexity and the need for ongoing management to mitigate DoS risks, there's a valid debate over whether the privacy benefits of Dandelion++ justify these costs. While privacy is a crucial aspect of the Bitcoin network, whether Dandelion++ is the most efficient way to achieve this goal is up for debate.
+
```

Comment:
lines 111 to 113 seem to have a different formatting style e.g. no line breaks between paragraphs. If this section was copied from the technical paper or other article you need to include quotes around the text and include the source after the section.

Reply:
i referenced the conversation in the link below